components/p3a/message_manager.cc

```diff
@@ -338,8 +350,18 @@ void MessageManager::StartScheduledConstellationPrep(MetricLogType log_type) {
   VLOG(2) << "MessageManager::StartScheduledConstellationPrep - Requesting "
              "randomness for histogram: "
           << log_key;
+
+  bool is_nebula = p3a::kNebulaOnlyHistograms.contains(log_key);
+  if (is_nebula && !features::IsNebulaEnabled()) {
+    // Do not report if Nebula feature is not enabled,
+    // mark request as successful to avoid transmission.
+    constellation_prep_log_stores_[log_type]->DiscardStagedLog();
+    constellation_prep_schedulers_[log_type]->UploadFinished(true);
+    delegate_->OnMetricCycled(log_key, true);
```

Comment:
shouldn't we return here?

Reply:
yes, good catch. thanks

Reply:
I added a return in 5da6c5fec7fb82ddf0f68c9c0d86225ae2cac32c.